File: engine/model/src/main/scala/tools/packet/party/PartyStatusMessage.scala
package tools.packet.party

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class PartyStatusMessage(private var _message: Int, private var _fromCharacterName: Option[String]) extends PacketInput {
  def message: Int = _message

  def fromCharacterName: Option[String] = _fromCharacterName

  def this(_message: Int) = {
    this(_message, Option.empty)
  }

  override def opcode(): SendOpcode = SendOpcode.PARTY_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/party/PartyStatusMessage.java
package tools.packet.party;

import java.util.Optional;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record PartyStatusMessage(Integer message, Optional<String> fromCharacterName) implements PacketInput {
   public PartyStatusMessage(Integer message) {
      this(message, Optional.empty());
   }

   @Override
   public SendOpcode opcode() {
      return SendOpcode.PARTY_OPERATION;
   }
}